tripper/dataset/dataset.py

```diff
@@ -0,0 +1,942 @@
+    shortnames = get_shortnames()
+    dct: dict = {}
+    for p, o in ts.predicate_objects(ts.expand_iri(iri)):
+        add(dct, shortnames.get(p, p), as_python(o))
+    if dct:
+        d = {"@id": iri}
+        d.update(dct)
+        return d
+    return dct
+
+
+def _load_sparql(ts: Triplestore, iri: str) -> dict:
+    """Load `iri` from triplestore by calling `ts.query()`."""
+    # The match-all pattern `(:|!:)*` in the query string below
+    # ensures that the returned triples includes nested structures,
+    # like distributions in a dataset. However, it does not include
+    # references to named resources, like parsers and generators.
+    # This is good, because it limits the number of returned triples.
+    # The `recur()` function will load such named resources recursively.
+    #
+    # Note that this implementation completely avoid querying for
+    # blank nodes, which avoids problems with backends that renames
+    # blank nodes.
+    subj = iri if iri.startswith("_:") else f"<{iri}>"
+    query = f"""
+    PREFIX : <http://example.com#>
+    CONSTRUCT {{ ?s ?p ?o }}
+    WHERE {{
+      {subj} (:|!:)* ?o .
+      ?s ?p ?o .
+    }}
+    """
+    nested = dicttypes.keys()
+
+    def recur(d):
+        """Recursively load and insert referred resources into dict `d`."""
+        if isinstance(d, dict):
+            for k, v in d.items():
+                if k in nested:
+                    val = v.get("@id")
+                    if isinstance(val, str):
+                        d[k] = load_dict(ts, val, use_sparql=True)
+                    elif isinstance(v, (dict, list)):
+                        recur(v)
+        elif isinstance(d, list):
+            for e in d:
+                recur(e)
+
+    triples = ts.query(query)
+    with Triplestore(backend="rdflib") as ts2:
+        ts2.add_triples(triples)  # type: ignore
+        dct = load_dict(ts2, iri, use_sparql=False)
+    recur(dct)
+    return dct
+
+
+def get_values(
+    data: "Union[dict, list]", key: str, extend: bool = True
+) -> list:
+    """Parse `data` recursively and return a list with the values
+    corresponding to the given key.
+
+    If `extend` is true, the returned list will be extended with
+    values that themselves are list, instead of appending them in a
+    nested manner.
+
+    """
+    values = []
+    if isinstance(data, dict):
+        val = data.get(key)
+        if extend and isinstance(val, list):
+            values.extend(val)
+        elif val:
+            values.append(val)
+        for v in data.values():
+            if isinstance(v, (dict, list)):
+                values.extend(get_values(v, key))
+    elif isinstance(data, list):
+        for ele in data:
+            if isinstance(ele, (dict, list)):
+                values.extend(get_values(ele, key))
+    return values
+
+
+@cache  # type: ignore
+def get_context(timeout: float = 5, fromfile: bool = True) -> dict:
+    """Return context as a dict.
```

Comment:
Which context is it getting? All context? Context defined how? This needs to be described better.